subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishExternalVariantIntegrationTest.groovy

```diff
@@ -0,0 +1,861 @@
+  - Variant firstRuntimeElements:
+      -  contains dependencies that will produce a pom file that cannot be consumed by a Maven client.
+          - Cannot determine variant coordinates for dependency 'org:foo' since the resolved graph does not contain the requested module.""")
+
+        def first = javaLibrary(mavenRepo.module('org', 'root-first', '1.0'))
+        def resolved = first.parsedPom.scopes.runtime.dependencies.values()
+        resolved*.artifactId == ["foo"]
+    }
+
+    def "fails if resolution fails"() {
+        given:
+        publishes(multiCoordinateComponent {
+            compilation("first")
+        })
+
+        buildFile << """
+            repositories { maven { url "${mavenRepo.uri}" } }
+            dependencies {
+                firstImplementation "org:foo:1.0"
+            }
+        """
+
+        when:
+        fails("publish")
+
+        then:
+        failure.assertHasCause("Could not find org:foo:1.0")
+
+        when:
+        fails("publish", "--stacktrace")
+
+        then:
+        // Not sure why this does not show up without stacktrace
+        failure.assertHasErrorOutput("Could not map coordinates for org:foo:1.0")
+    }
+
+    // This simulates the way AGP and KGP publish
+    @NotYetImplemented
+    def "publishes resolved child coordinates for multi-coordinate project dependency when target component uses separate local and published configurations"() {
+        given:
+        publishes(multiCoordinateComponent {
+            compilation("first") {
+                withSeparatePublishedConfiguration = true
+            }
+            compilation("second") {
+                withSeparatePublishedConfiguration = true
+            }
+        })
+
+        buildFile << """
+            dependencies {
+                firstImplementation project(':other')
+                secondImplementation project(':other')
+            }
+        """
+
+        when:
+        succeeds(":publish")
+
+        then:
+        def root = javaLibrary(mavenRepo.module('org', 'root', '1.0'))
+        def first = javaLibrary(mavenRepo.module('org', 'root-first', '1.0'))
+        def second = javaLibrary(mavenRepo.module('org', 'root-second', '1.0'))
+
+        !root.parsedPom.scopes.compile
+        !root.parsedPom.scopes.runtime
+
+        def firstDeps = first.parsedPom.scopes.runtime.dependencies
+        firstDeps.size() == 1
+        def firstDep = firstDeps.values().first()
+        firstDep.groupId == "org"
+        firstDep.artifactId == "other-first"
+        firstDep.version == "1.0"
+
+        def secondDeps = second.parsedPom.scopes.runtime.dependencies
+        secondDeps.size() == 1
+        def secondDep = secondDeps.values().first()
+        secondDep.groupId == "org"
+        secondDep.artifactId == "other-second"
+        secondDep.version == "1.0"
+    }
+
+    // region Component builder fixture
```

Comment:
🤔 I think without our discussion this fixture would have been very difficult to understand.  It would be great to have an overall present in the code here.

Reply:
What do you mean by `overall present`?

Reply:
Wow, that's a bad autocorrection.  I believe I was trying to say have a "roadmap present".

Reply:
Added some context